spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/Query.java

```diff
@@ -44,6 +45,13 @@
 
 	/**
 	 * Optional {@link RowMapper} to use to convert the result of the query to domain class instances.
+	 * Cannot be used along with {@link #resultSetExtractorClass()} only one of the two can be set.
 	 */
 	Class<? extends RowMapper> rowMapperClass() default RowMapper.class;
+	
+	/**
+	 * Optional {@link ResultSetExtractor} to use to convert the result of the query to domain class instances.
+	 * Cannot be used along with {@link #rowMapperClass()} only one of the two can be set.
+	 */
+	Class<? extends ResultSetExtractor> resultSetExtractorClass() default ResultSetExtractor.class;
```

Comment:
I think we should refine the general approach. Result extraction uses by default `RowMapperResultSetExtractor` which accepts a `RowMapper`. It would make sense to expose this detail and adjust the underlying assumption.  See general review comment.